assets/javascripts/discourse-assign/controllers/assign-user.js.es6

```diff
@@ -77,48 +107,47 @@ export default Controller.extend({
 
   @action
   assign() {
+    this.reassignOrAssignTarget("assign");
+  },
+
+  @action
+  reassign() {
+    this.reassignOrAssignTarget("reassign");
+  },
+
+  @action
+  reassignUser(name) {
     if (this.isBulkAction) {
-      this.bulkAction(this.model.username);
+      this.bulkAction(name);
       return;
     }
-    let path = "/assign/assign";
 
-    if (isEmpty(this.get("model.username"))) {
-      this.model.target.set("assigned_to_user", null);
-    }
-
-    if (isEmpty(this.get("model.group_name"))) {
-      this.model.target.set("assigned_to_group", null);
+    if (this.allowedGroupsForAssignment.includes(name)) {
+      this.setProperties({
```

Comment:
generally speaking I would prefer: this.model.set... but that's a nitpick feel free to ignore.

The reasoning is that you want to be sure you have a model here, your current code will hide any issue where model actually doesn’t exist.